client/python/rappor.py

```diff
@@ -119,39 +119,37 @@ def from_csv(f):
     return p
 
 
-class _SimpleRandom(object):
+class _SecureRandom(object):
   """Returns an integer where each bit has probability p of being 1."""
 
-  def __init__(self, prob_one, num_bits, _rand=None):
+  def __init__(self, prob_one, num_bits):
     self.prob_one = prob_one
     self.num_bits = num_bits
-    self._rand = _rand or random.Random()
 
   def __call__(self):
     p = self.prob_one
-    rand_fn = self._rand.random  # cache it for speed
+    bytes = os.urandom(self.num_bits)
 
     r = 0
-    for i in xrange(self.num_bits):
-      bit = rand_fn() < p
+    for i, b in enumerate(xrange(self.num_bits)):
+      bit = b < p * 255.0
```

Comment:
Ah I see your point. I was just thinking about it in terms of "we don't care about P enough that 0.3984375 is unacceptable", which is true. But if the analysis thinks it is .39999999... and the client reports it using 0.3984375 then we are going to be skewing our results. Yes, we should work out some way for our clients to agree with the analysis exactly on what the probabilities are. We could do this by limiting the valid parameters. But that is kindof annoying...